lib/usdUfe/ufe/UsdAttributeHolder.cpp

```diff
@@ -59,17 +61,58 @@ bool setUsdAttrMetadata(
 
     UsdUfe::AttributeEditRouterContext ctx(attr.GetPrim(), attr.GetName());
 
+    // These metadata have a USD equivalent. Use it to stay compatible with other DCCs:
+    static const auto knownUSDMetadata
+        = std::unordered_set<std::string> { UsdUfe::MetadataTokens->UIDoc,
+                                            UsdUfe::MetadataTokens->UIEnumLabels,
+                                            UsdUfe::MetadataTokens->UIName,
+                                            UsdUfe::MetadataTokens->UIFolder };
+
     PXR_NS::TfToken tok(key);
     if (PXR_NS::UsdShadeNodeGraph(attr.GetPrim())) {
         if (PXR_NS::UsdShadeInput::IsInput(attr)) {
             PXR_NS::UsdShadeInput input(attr);
-            input.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                input.ClearSdrMetadataByKey(tok);
```

Comment:
I guess this is for the case where we put the metadata in older version of MayaUSD and we want to clean it up now.

Reply:
Yes, we want it as close to USD specs as possible, so we will move it from SdrMetadata to Sdf if there was old metadata from a previously saved file.